test/e2e/util.go

```diff
@@ -571,6 +536,68 @@ func RunRC(c *client.Client, name string, ns, image string, replicas int) error
 	return nil
 }
 
+func ResizeRC(c *client.Client, ns, name string, size int) error {
+	By(fmt.Sprintf("Resizing replication controller %s to %d", name, size))
+	rc, err := c.ReplicationControllers(ns).Get(name)
+	if err != nil {
+		return fmt.Errorf("Failed to find replication controller %s in namespace %s: %v", name, ns, err)
+	}
+	rc.Spec.Replicas = size
+	if _, err := c.ReplicationControllers(ns).Update(rc); err != nil {
+		return fmt.Errorf("Failed to resize replication controller %s: %v", name, err)
+	}
+
+	// Wait up to 10 minutes until all replicas are killed.
+	endTime := time.Now().Add(time.Minute * 10)
+	for {
+		if time.Now().After(endTime) {
+			return fmt.Errorf("Timeout while waiting for replication controller %s replicas to %s", name, size)
+		}
+		remainingTime := endTime.Sub(time.Now())
+		err := wait.Poll(time.Second, remainingTime, client.ControllerHasDesiredReplicas(c, rc))
+		if err != nil {
+			Logf("Error while waiting for replication controller %s replicas to read %v: %s", name, size, err)
+		} else {
+			break
+		}
+	}
+
+	// Wait up to 10 minutes for pods to become Running.
+	running := false
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	for start := time.Now(); time.Since(start) < 10*time.Minute; time.Sleep(5 * time.Second) {
```

Comment:
Listing all pods every 5s was exactly the thing a lot of our work was done to avoid, since it places quite a bit of load on the system by itself :)

Unsure how much context you have about similar code, but another way to achieve this would be to setup a reflector, eg: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/config/apiserver.go#L43. A controller is an even higher abstraction, but I think a reflector should do for this. You can then poll the store.